core/unit_test/TestBitManipulationBuiltins.hpp

```diff
@@ -789,7 +789,14 @@ struct TestBitCastFunction {
     ASSERT_EQ(errors, 0) << "Failed check no error for bit_cast()";
   }
   template <typename To, typename From>
-  static KOKKOS_FUNCTION bool check(const From& from) {
+#if defined(KOKKOS_COMPILER_GNU) && (910 <= KOKKOS_COMPILER_GNU) && \
```

Comment:
I tested SHA 6b8f4736d41eef8e2e66fd2455efd4c115e2124f with the systems gcc/9.1 module, which turned out to be misnamed and really should have been gcc/9.0.1

@dalg24 With gcc/9.0.1 the compilation error is present, can the compiler check be updated to guard for gcc/9.0 as well?

```
[ndellin@kokkos-dev-2 Gcc9.1]$ g++ --version
g++ (GCC) 9.0.1 20190428 (prerelease)
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

```
/ascldap/users/ndellin/kokkos/core/unit_test/TestBitManipulationBuiltins.hpp: In member function ‘void TestBitCastFunction<Space>::operator()(int, int&) const’:
/ascldap/users/ndellin/kokkos/core/unit_test/TestBitManipulationBuiltins.hpp:819:29: error: no matching function for call to ‘TestBitCastFunction<Space>::check<int>(float) const’
  819 |       if (!check<int>(12.34f)) {
      |                             ^

```

Including `900` in the guard resolved the compilation error (I'm assuming the error will be present on 9.0.0 as well)

```suggestion
#if defined(KOKKOS_COMPILER_GNU) && (900 <= KOKKOS_COMPILER_GNU) && \
```